src/components/sidebar/Sidebar.js

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from 'react';
+import { fetchAlllUsers } from '../../network/api-crud';
+import './sidebar.css';
+
+export default function Sidebar(props) {
+  const [users, setUsers] = useState([]);
+  const handleClick = (userData) => {
+    props.user({ username: userData.username, password: userData.password })
+  }
+
+  useEffect(() => {
+    fetchAlllUsers((data) => {
```

Comment:
instead of passing function as callback we can directly return promises and use the .then .catch here
this will also help in handling the error case

Reply:
done